lib/stitches/tangential_fill_stitch_pattern_creator.py

```diff
@@ -0,0 +1,912 @@
+                        result_coords.append(point)
+                        result_coords_origin.append(child_coords_origin[0])
+
+                    result_coords.extend(child_coords[1:])
+                    result_coords_origin.extend(child_coords_origin[1:])
+                else:
+                    result_coords.extend(child_coords)
+                    result_coords_origin.extend(child_coords_origin)
+
+                # And here is the point for the leaving of the child
+                # (distance to the own following point should not be too large)
+                d = item.nearest_point_parent.distance(Point(own_coords[i]))
+                if cur_item < len(nearest_points_list) - 1:
+                    d = min(
+                        d,
+                        abs(nearest_points_list[cur_item + 1].proj_distance_parent - item.proj_distance_parent),
+                    )
+
+                if d > constants.factor_offset_starting_points * abs_offset:
+                    result_coords.append(
+                        current_coords.interpolate(item.proj_distance_parent + 2 * constants.factor_offset_starting_points * abs_offset).coords[0]
+                    )
+                    result_coords_origin.append(
+                        sample_linestring.PointSource.ENTER_LEAVING_POINT
+                    )
+                    # Check whether this additional point makes the last point
+                    # of the child unnecessary
+                    point = calculate_replacing_middle_point(
+                        LineString(
+                            [result_coords[-3], result_coords[-2], result_coords[-1]]
+                        ),
+                        abs_offset,
+                        stitch_distance,
+                    )
+                    if point is None:
+                        result_coords.pop(-2)
+                        result_coords_origin.pop(-2)
+
+                cur_item += 1
+            if i < len(own_coords) - 1:
+                if (Point(result_coords[-1]).distance(Point(own_coords[i])) > abs_offset * constants.factor_offset_remove_points):
+                    result_coords.append(own_coords[i])
+                    result_coords_origin.append(own_coords_origin[i])
+
+            # Since current_coords and own_coords are rastered differently
+            # there accumulate errors regarding the current distance.
+            # Since a projection of each point in own_coords would be very
+            # time consuming we project only every n-th point which resets
+            # the accumulated error every n-th point.
+            if i % 20 == 0:
+                total_distance = current_coords.project(Point(own_coords[i]))
+            else:
+                total_distance += next_distance
+
+    assert len(result_coords) == len(result_coords_origin)
+    return result_coords, result_coords_origin
+
+
+# Partly taken from https://github.com/mikedh/pocketing/blob/master/pocketing/polygons.py
```

Comment:
In order to use the code from pocketing like this, we probably have to include its BSD license and copyright statement.  I'd like to avoid doing that if possible because it complicates our own license, which is GPL.

Fortunately, I don't think we'll need to worry about that if we implement my idea to get rid of the KD tree parts, since we'll end up with a function that is substantially different.  At that point we'll just be using pocketing's code as inspiration rather than directly copying.